### PR TITLE
SFR-1588:  Add Collection page

### PR DESCRIPTION
**Pull request by jackiequach**
[Jira Ticket](http://jira.nypl.org/browse/SFR-1588)

### This PR does the following:
- Add [collectionId] page endpoint and corresponding Collection component

### Testing requirements & instructions: 
-

### CODE REVIEW

File: src/util/CollectionUtils.tsx [hunk 1]
```diff
@@ -24,4 +34,208 @@ export default class CollectionUtils {
...
+            Request
+          </Link>
+        </>
+      );
+    } else {
+      return (
+        <>
+          May be available via NYPL<br></br>
+          <Link
+            to={`https://login.nypl.org/auth/login?redirect_uri=${encodeURIComponent(
```
Comment (samanthaandrews) [inline src/util/CollectionUtils.tsx:230]: Nit: consider moving this login url out into a constant

File: src/util/CollectionUtils.tsx [hunk 2]
```diff
@@ -24,4 +34,208 @@ export default class CollectionUtils {
...
+
+  static getEddLinkElement(eddLink: OpdsLink, isLoggedIn: boolean) {
+    if (isLoggedIn) {
+      return (
+        <>
+          <Box whiteSpace="initial">
+            You can request a partial scan via NYPL
+          </Box>
+          <Link
+            to="https://www.nypl.org/research/scan-and-deliver"
```
Comment (samanthaandrews) [inline src/util/CollectionUtils.tsx:210]: Nit: consider moving this login url out into a constant

File: src/util/CollectionUtils.tsx [hunk 3]
```diff
@@ -24,4 +34,208 @@ export default class CollectionUtils {
...
+        </Link>
+      );
+    }
+
+    return undefined;
+  };
+
+  static getNoLinkElement(showRequestButton: boolean) {
+    if (showRequestButton) {
+      return <span>Not Yet Available {showRequestButton}</span>;
```
Comment (samanthaandrews) [inline src/util/CollectionUtils.tsx:134]: Does this render a boolean value of true or false?

File: src/util/CollectionUtils.tsx [hunk 4]
```diff
@@ -24,4 +34,208 @@ export default class CollectionUtils {
...
+    }
+    return <>Languages: Undetermined</>;
+  }
+
+  // Rights
+  static getLicense(rights: Rights) {
+    return rights ? `License: ${rights.rightsStatement}` : "License: Unknown";
+  }
+
+  static getReadLink = (links: OpdsLink[], type: "readable" | "embedable") => {
```
Comment (samanthaandrews) [inline src/util/CollectionUtils.tsx:103]: Since it seems like the `type: "readable" | "embedable"` is specific to what we can read/open in this app, we might want to make is a named type that we can reuse. Something like `export type ReadOnlineTypes = "readable" | "embedable"`

File: src/components/Collection/Collection.tsx [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+import CollectionUtils from "~/src/util/CollectionUtils";
+import ResultsSorts from "../ResultsSorts/ResultsSorts";
+import { useRouter } from "next/router";
+import { CollectionItemCard } from "../CollectionItemCard/CollectionItemCard";
+import Link from "../Link/Link";
+
+const Collection: React.FC<{
+  collectionQuery: CollectionQuery;
+  collectionResult: CollectionResult;
+}> = (props) => {
```
Comment (samanthaandrews) [inline src/components/Collection/Collection.tsx:42]: Nit: we can destructure props here, i.e. `({ collectionResult, collectionQuery })` and then further destructure `collectionResult` within the component.

**Review by samanthaandrews — CHANGES_REQUESTED** [review]
I left a couple comments! Overall I need to see two things:

1. Convert functions that return JSX that are called directly inside a component into React functional components and render them directly.
2. Unit tests for sorting and pagination. Ideally we could also do this with integration tests using Cypress or Playwright, but for now, let's create unit tests. Specifically let's make sure we cover potential edge cases.

File: src/components/Collection/Collection.test.tsx [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+describe("Renders Collection Page", () => {
+  beforeEach(() => {
+    render(
+      <Collection
+        collectionQuery={collectionQuery}
+        collectionResult={collectionResults}
+      />
+    );
+    act(() => {
+      resizeWindow(300, 1000);
```
[the remark this answers is not in the document]
Comment (jackiequach) [reply src/components/Collection/Collection.test.tsx:29]: Copied this from another test file, it was being used for mobile view. Removed in #389

File: src/util/CollectionUtils.tsx (hunk 3 above)
Comment (jackiequach) [reply src/util/CollectionUtils.tsx:86]: Copied from `EditionCardUtils`, but looks like `showRequestButton` is never true. Removed for this.

File: src/util/CollectionUtils.tsx [hunk 7]
```diff
@@ -24,4 +34,208 @@ export default class CollectionUtils {
...
+  };
+
+  static getNoLinkElement(showRequestButton: boolean) {
+    if (showRequestButton) {
+      return <span>Not Yet Available {showRequestButton}</span>;
+    }
+    return <>Not yet available</>;
+  }
+  static selectDownloadLink = (links: OpdsLink[]) => {
+    if (links) return undefined;
```
[the remark this answers is not in the document]
Comment (jackiequach) [reply src/util/CollectionUtils.tsx]: Yes, good catch.

**Review by samanthaandrews — CHANGES_REQUESTED** [review]
Hey @jackiequach! I started reviewing this again but realized that I would be able to articulate what I'm looking for by coding it up in a separate branch, which now has an open PR here https://github.com/NYPL/sfr-bookfinder-front-end/pull/390/files

That PR has a lot of failing checks, which makes sense because I refactored it quickly without actually running the code 😅 I'm currently unable to run this project locally at all (I think due to the fact that I'm on a mac with an M1 chip).

I'm thinking it makes sense for you to pull down that branch and see if you can run it, get the tests to pass, and explore the refactoring that I did. The big takeaway is that I converted anything that renders JSX into a React component, and kept everything in CollectionUtils to only include functions that return text, OpdsLinks, etc. that would eventually get rendered inside a component. I think this makes the code easier to reason about and easier to test.

I also pulled out the various components that were part of CollectionItemCard into their own files, as that component was getting large and it needed to be more readable. This is more of a stylistic preference on my end.

File: src/components/CollectionList/CollectionList.tsx [hunk 8]
```diff
@@ -13,9 +13,15 @@ export const CollectionList: React.FC<{ collections: Opds2Feed }> = ({
           <SimpleGrid gap="l" columns={4}>
             {collections.groups.map((collection, index) => {
               if (index < 8)
```
Comment (samanthaandrews) [inline src/components/CollectionList/CollectionList.tsx:15]: Let's turn this magic number 8 into a named constant so we know why it is specifically set to 8

File: src/components/Collection/Collection.tsx [hunk 9]
```diff
@@ -0,0 +1,245 @@
...
+          backgroundColor="section.research.primary"
+          heroType="tertiary"
+          heading={
+            <Heading level="one" id="tertiary-hero">
+              <>
+                Digital Research Books <sup>Beta</sup>
+              </>
+            </Heading>
+          }
+        />
```
Comment (samanthaandrews) [inline src/components/Collection/Collection.tsx:162]: Seems like this is a component that is duplicated in a couple of other places in the code. Let's pull it out into a separate component and re-use it in all the places it belongs.

File: src/components/Collection/Collection.tsx [hunk 10]
```diff
@@ -0,0 +1,245 @@
...
+                  {pub.metadata.subtitle && (
+                    <Box marginBottom="xs">{pub.metadata.subtitle}</Box>
+                  )}
+                  {pub.metadata.creator && (
+                    <Box marginBottom="xs">
+                      By <Author author={pub.metadata.creator} />{" "}
+                    </Box>
+                  )}
+                  <CollectionItemCard collectionItem={pub}></CollectionItemCard>
+                </Box>
```
Comment (samanthaandrews) [inline src/components/Collection/Collection.tsx:226]: Since this component is quite large and complex, let's pull this out into a separate component and put it in a new file.

**Review by samanthaandrews — APPROVED** [review]
